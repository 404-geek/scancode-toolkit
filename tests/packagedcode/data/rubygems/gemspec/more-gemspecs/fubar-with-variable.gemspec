

some_variable = "fubar"

Gem::Specification.new do |spec|
  spec.name         = some_variable
  spec.version      = "0.4.2"
  spec.summary      = "Fine Uber Archive"
  spec.description  = "A archive extraction library"

  spec.author       = "nexB"
  spec.email        = "info@nexb.com"
  spec.homepage     = "https://github.com/pombredanne/fubar"
  spec.license      = "ISC"

  spec.require_path = "lib"
  spec.files        = Dir["CHANGELOG.md", "LICENSE.md", "README.md",
                          "lib/**/*"]

  spec.required_ruby_version = ">= 2.4.1"
  spec.required_rubygems_version = ">= 2.6.12"

  spec.add_dependency 'foo', '1.0.0'
  spec.add_dependency 'bar', '= 1.0.0'
end

