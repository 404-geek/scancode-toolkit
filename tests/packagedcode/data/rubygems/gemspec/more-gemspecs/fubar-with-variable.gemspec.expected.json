[
  {
    "type": "gem",
    "namespace": null,
    "name": "some_variable",
    "version": "0.4.2",
    "qualifiers": {},
    "subpath": null,
    "primary_language": "Ruby",
    "description": "Fine Uber Archive\nA archive extraction library",
    "release_date": null,
    "parties": [
      {
        "type": "person",
        "role": "author",
        "name": "nexB",
        "email": null,
        "url": null
      },
      {
        "type": "person",
        "role": "author",
        "name": null,
        "email": "info@nexb.com",
        "url": null
      }
    ],
    "keywords": [],
    "homepage_url": "https://github.com/pombredanne/fubar",
    "download_url": "https://rubygems.org/downloads/some_variable-0.4.2.gem",
    "size": null,
    "sha1": null,
    "md5": null,
    "sha256": null,
    "sha512": null,
    "bug_tracking_url": null,
    "code_view_url": null,
    "vcs_url": null,
    "copyright": null,
    "declared_license_expression": "isc",
    "declared_license_expression_spdx": "ISC",
    "license_detections": [
      {
        "license_expression": "isc",
        "detection_log": [
          "not-combined"
        ],
        "matches": [
          {
            "score": 100.0,
            "start_line": 1,
            "end_line": 1,
            "matched_length": 1,
            "match_coverage": 100.0,
            "matcher": "1-spdx-id",
            "license_expression": "isc",
            "rule_identifier": "spdx-license-identifier: isc",
            "rule_url": null,
            "referenced_filenames": [],
            "is_license_text": false,
            "is_license_notice": false,
            "is_license_reference": false,
            "is_license_tag": true,
            "is_license_intro": false,
            "rule_length": 1,
            "rule_relevance": 100,
            "matched_text": "ISC",
            "licenses": [
              {
                "key": "isc",
                "name": "ISC License",
                "short_name": "ISC License",
                "category": "Permissive",
                "is_exception": false,
                "is_unknown": false,
                "owner": "ISC - Internet Systems Consortium",
                "homepage_url": "https://www.isc.org/licenses/",
                "text_url": "http://fedoraproject.org/wiki/Licensing:MIT#Old_Style_with_legal_disclaimer_2",
                "reference_url": "https://scancode-licensedb.aboutcode.org/isc",
                "scancode_url": "https://github.com/nexB/scancode-toolkit/tree/develop/src/licensedcode/data/licenses/isc.LICENSE",
                "spdx_license_key": "ISC",
                "spdx_url": "https://spdx.org/licenses/ISC"
              }
            ]
          }
        ]
      }
    ],
    "other_license_expression": null,
    "other_license_expression_spdx": null,
    "other_license_detections": [],
    "extracted_license_statement": "ISC",
    "notice_text": null,
    "source_packages": [],
    "file_references": [],
    "extra_data": {},
    "dependencies": [
      {
        "purl": "pkg:gem/foo",
        "extracted_requirement": "1.0.0",
        "scope": "dependency",
        "is_runtime": true,
        "is_optional": false,
        "is_resolved": false,
        "resolved_package": {},
        "extra_data": {}
      },
      {
        "purl": "pkg:gem/bar",
        "extracted_requirement": "= 1.0.0",
        "scope": "dependency",
        "is_runtime": true,
        "is_optional": false,
        "is_resolved": false,
        "resolved_package": {},
        "extra_data": {}
      }
    ],
    "repository_homepage_url": "https://rubygems.org/gems/some_variable/versions/0.4.2",
    "repository_download_url": "https://rubygems.org/downloads/some_variable-0.4.2.gem",
    "api_data_url": "https://rubygems.org/api/v2/rubygems/some_variable/versions/0.4.2.json",
    "datasource_id": "gemspec",
    "purl": "pkg:gem/some_variable@0.4.2"
  }
]